src/types.ts

```diff
@@ -1,44 +1,39 @@
-export type VoiceStartMessage = {
-  type: 'voiceStart';
+export type MeasureStartMessage = {
+  msgType: 'measureStart';
+  width: string;
+  staves: string;
+  clef: string[];
+  time: string;
 };
 
-export type VoiceEndMessage = {
-  type: 'voiceEnd';
+export type MeasureEndMessage = {
+  msgType: 'measureEnd';
 };
 
 export type BeamStartMessage = {
-  type: 'beamStart';
+  msgType: 'beamStart';
 };
 
 export type BeamEndMessage = {
-  type: 'beamEnd';
+  msgType: 'beamEnd';
 };
 
 export type NoteMessage = {
-  type: 'note';
+  msgType: 'note';
+  rest: boolean;
```

Comment:
I know that EasyScore treats rests as notes, but we should just create a separate `RestMessage`. Most of the properties in `NoteMessage` are not applicable to rests.

Reply:
As said before I would try to keep MusicXML approach and there, rest is an attribute of note.